### PR TITLE
DC-72: TagMatchers: Query only on node level, not above

### PR DESCRIPTION
**Pull request by patrick-schweizer**
* JIRA (Issue Tracker): https://issues.opennms.org/browse/DC-72

### CODE REVIEW

File: features/timeseries/src/main/java/org/opennms/netmgt/timeseries/samplewrite/TimeseriesWriter.java [hunk 1]
```diff
@@ -161,12 +161,9 @@ public void destroy() {
     }
 
     public void insert(List<Sample> samples) {
```
Comment (patrick-schweizer) [inline features/timeseries/src/main/java/org/opennms/netmgt/timeseries/samplewrite/TimeseriesWriter.java:163]: this is just to simplify code, has nothing to do with the querying

File: features/timeseries/src/main/java/org/opennms/netmgt/timeseries/util/TimeseriesUtils.java [hunk 2]
```diff
@@ -47,9 +47,14 @@
...
-
-    public static final int WILDCARD_INDEX_NO = 2; // => node level
+    /**
+     * This constant defines at which level in a resource path we start to query for multiple metrics.
+     * To save queries, we want to get all resources under a node.
+     * requisitions have the node at the 4th position, example: snmp/fs/xx/4/
+     * auto detected metrics have the node at 2nd position, example: snmp/4/mib2-tcp
+     * since we can't distinguish we set the level to 4.
+     */
+    public static final int WILDCARD_INDEX_NO = 4;
```
Comment (patrick-schweizer) [inline features/timeseries/src/main/java/org/opennms/netmgt/timeseries/util/TimeseriesUtils.java:57]: I wonder if we can distinguish between the 2 types of resources (provision vs. auto detect).
Maybe check if the second or 4th element is a number? Not sure if that would work?

Comment (patrick-schweizer) [reply features/timeseries/src/main/java/org/opennms/netmgt/timeseries/util/TimeseriesUtils.java:57]: @j-white what is your take on this?

**Review by j-white — APPROVED** [review]

**Review by agalue — APPROVED** [review]
I believe the change is better and more constrained than the original approach, therefore a lot faster.

However, based on my tests, I can see subsequent queries for each IP interface on the node, which introduces other challenges, but I believe it is worth having the change in place even with them.

That said, I'd love to hear opinions from other OpenNMS developers.

**Review by j-white — APPROVED** [review]

**Pull request merged**